### PR TITLE
hw-3

### PR DESCRIPTION
**Pull request by AndreiYazadzhi**

### CODE REVIEW

**Review by MykolaFatkullin — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -8,5 +8,61 @@
...
+        <project.reporting.outputEncoding>UTF-8</project.reporting.outputEncoding>
+        <maven.checkstyle.plugin.version>3.1.1</maven.checkstyle.plugin.version>
+        <maven.checkstyle.plugin.configLocation>
+            https://raw.githubusercontent.com/mate-academy/style-guides/master/java/checkstyle.xml
+        </maven.checkstyle.plugin.configLocation>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>junit</groupId>
```
Comment (Sofasmile) [inline pom.xml]: do we need this dependency?

File: pom.xml [hunk 2]
```diff
@@ -8,5 +8,61 @@
...
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 
 </project>
```
Comment (Sofasmile) [inline pom.xml]: let's add an emty line at the end of the file

File: src/main/java/com/dao/homework/dao/ManufacturerDao.java [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    Optional<Manufacturer> getById(Long manufacturerId);
+
+    Manufacturer update(Manufacturer manufacturer);
+
+    boolean delete(Manufacturer manufacturer);
+
+    boolean deleteById(Long manufacturerId);
+
+    List<Manufacturer> getAllManufacturers();
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/dao/ManufacturerDao.java]:
```suggestion
    List<Manufacturer> getAll();
```

File: src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+                .findFirst();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturer) {
+        if (!getAllManufacturers().contains(manufacturer)) {
+            throw new RuntimeException("Can`t find this element!");
+        }
+        manufacturer.setId(manufacturer.getId());
+        manufacturer.setCountry(manufacturer.getCountry());
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java]: this code isn't flexible. We should change the logic of this method if the manufacturer object is changed
let's use the method `set` from the list for replacing an object

File: src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java [hunk 5]
```diff
@@ -0,0 +1,57 @@
...
+        return manufacturer;
+    }
+
+    @Override
+    public boolean delete(Manufacturer manufacturer) {
+        if (!getAllManufacturers().contains(manufacturer)) {
+            throw new RuntimeException("Can`t find this element!");
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java]:
```suggestion
            return false;
```

File: src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+        }
+        Storage.manufacturers.remove(manufacturer.getId().intValue());
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java]: does this method work correctly?

File: src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+        return true;
+    }
+
+    @Override
+    public boolean deleteById(Long manufacturerId) {
+        if (!getAllManufacturers().contains(getById(manufacturerId).get())) {
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/dao/ManufacturerDaoImpl.java]: read about method `removeIf`

File: src/main/java/com/dao/homework/service/ManufacturerService.java [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+
+import com.dao.homework.model.Manufacturer;
+import java.util.List;
+
+public interface ManufacturerService {
+    Manufacturer create(Manufacturer manufacturer);
+
+    Manufacturer get(Long id);
+
+    List<Manufacturer> getAllManufacturers();
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/service/ManufacturerService.java]:
```suggestion
    List<Manufacturer> getAll();
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dao/homework/dao/ManufacturerDao.java [hunk 9]
```diff
@@ -11,9 +11,7 @@
 
     Manufacturer update(Manufacturer manufacturer);
 
-    boolean delete(Manufacturer manufacturer);
-
-    boolean deleteById(Long manufacturerId);
+    boolean delete(Long manufacturerId);
 
     List<Manufacturer> getAllManufacturers();
```
Comment (Sofasmile) [inline src/main/java/com/dao/homework/dao/ManufacturerDao.java]: how about this comment?

**Review by Sofasmile — CHANGES_REQUESTED** [review]

File: src/main/java/com/dao/homework/dao/ManufacturerDao.java (hunk 9 above)
Comment (AndreiYazadzhi) [reply src/main/java/com/dao/homework/dao/ManufacturerDao.java]: Sorry,at first time didn`t understand problem.Fixed

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**